bench.py

```diff
@@ -0,0 +1,416 @@
+#
+# Usage: this program takes the following arguments: --mode, --directory, --filter, --outputDirectory, --endBufSize
+# See the main function for more details about the arguments.
+# Example command line: python bench.py --mode run --directory sccl-presynth/sccl_presynth --filter *gather.n16* --output output
+#
+# Details on "--mode test":
+# The "--mode test" option is used to test the program without actually running the benchmarks.
+# It will still iterate overall of the files in the directory and filter but it will not run the benchmarks.
+# The file contents do not matter as the variable exampleMCCLInput is used instead of the file contents.
+# The output is also faked with the results being the variable exampleOutput.
+#
+# TODO comments are scattered throughout the code to indicate areas that need improvement.
+#   Primarily these are hard-coded constants that should probably be command line arguments.
+#   Such as the separator for the CSV file.
+#
+def main():
+    # Setup an argument parser to parse the command line arguments
+    parser = argparse.ArgumentParser(description='Run NCCL benchmarks tests using NCCL and MSCCL')
+    parser.add_argument('-m', '--mode', type=str, help='mode: is either "test" or "run" where "test" indicates that the program should run in test mode and "run" indicates that the program should actually run the benchmarks.', default="run", choices=["test", "run"])
+    parser.add_argument('-d', '--directory', type=str, help='directory: is the directory where the MSCCL xml files are located', default="sccl-presynth/sccl_presynth")
+    parser.add_argument('-f', '--filter', type=str, help='filter: is a filter in the style of https://docs.python.org/3/library/fnmatch.html for files in the directory that will be tested', required=True)
+    parser.add_argument('-o', '--outputDirectory', type=str, help='outputDirectory: is the directory where the output files will be placed', required=True)
+    parser.add_argument('-e', '--endBufSize', type=str, help='endBufSize: is the end buffer size to use for the test. It is optional and defaults to 32MB.', default="32MB")
+
+    # Parse the command line arguments
+    args = parser.parse_args()
+
+    # Make sure sufficient command line arguments are provided
+    # and make sure the mode is either "test" or "run"
+    # if len(sys.argv) != 5 or (sys.argv[1] != "test" and sys.argv[1] != "run") :
+    #    print("Please provide line arguments: mode (test | run), directory, filter, outputDirectory")
+    #    return
+
+    mode = args.mode
+    directory = args.directory
+    filter = args.filter
+    outputDirectory = args.outputDirectory
+    endBufSize = args.endBufSize
+
+    # Print all of the command line arguments
+    print(f"Arguments: {args}")
+
+    # Separator to use in the CSV result file
+    # TODO: consider making this a command line argument
+    separator = "\t"
+
+    # This is the CSV file with the final results
+    resultFileName = "results.txt"
```

Comment:
I'm new to Python and realize I haven't been consistent in following a naming convention for variables.  I'll update these to be _ separated.